pkg/docker/docker.go

```diff
@@ -768,19 +892,21 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 		os.Exit(2)
 	}
 	return interrupt.New(dumpStack, removeContainer).Run(func() error {
-		// Attach to the container.
+		var attachError error
+		// Attach to the container on go thread (different than with fsouza, since it provided a non-blocking attach which we don't seem to have with k8s/engine-api)
+		// still preserve the flow of attaching before starting to handle various timing issues encountered in the past, as well as allow for --run option
 		glog.V(2).Infof("Attaching to container %q ...", containerName)
-		attachOpts := opts.asDockerAttachToContainerOptions()
-		attachOpts.Container = container.ID
-		if _, err = d.client.AttachToContainerNonBlocking(attachOpts); err != nil {
-			glog.V(0).Infof("error: Unable to attach to container %q with options %+v: %v", containerName, attachOpts, err)
-			return err
-		}
+		go func() {
+			attachOpts := opts.asDockerAttachToContainerOptions()
+			if attachError = d.kubeDockerClient.AttachToContainer(container.ID, attachOpts, opts.asDockerAttachToStreamOptions()); attachError != nil {
+				glog.V(0).Infof("error: Unable to attach to container %q with options %+v: %v", containerName, attachOpts, attachError)
+			}
+		}()
```

Comment:
Actually, what I feared earlier sort of does in fact transpire, even with select on error/timer channels, like is done in both util.TimeoutAfter and in pkg/tar/tar.go, unless we re-swizzle what is done on timeout.

My testing confirms that we always end up timing out because since the attach blocks, and the container never *exits* because it is created, not started, we always end up exiting on the timer channel.

I think the best we can do is say use our default, quick action, default docker timeout of 20 seconds, and if the error hasn't occurred by then, assume the attach is good, and move onto the 
start.